debug_module.adoc

```diff
@@ -328,12 +328,15 @@ that case, the command will return "not supported".
 ====
 
 Debuggers execute abstract commands by writing them to {dm-command}. They can
-determine whether an abstract command is complete by reading {abstractcs-busy} in {dm-abstractcs}. If the debugger starts a new command while {abstractcs-busy} is set, {abstractcs-cmderr} becomes 1 (busy), the
-currently executing command still gets to run to completion, but any
-error generated by the currently executing command is lost. After
+determine whether an abstract command is complete by reading {abstractcs-busy} in {dm-abstractcs}. After
 completion, {abstractcs-cmderr} indicates whether the command was successful or not.
```

Comment:
This part is a bit misleading -- `abstractcs.cmderr` does not indicate whether the completed command was successful or not if another command was written while `abstractcs.busy` was high. The command may have been successful but `abstractcs.cmderr` is not `none`.

Maybe something like:
`After completion, {abstractcs-cmderr} equal to 0 (none) indicates that the command was successful.`

Reply:
I understand your point and thank you for the suggestion. Still, I would like to leave this sentence as is for these reasons:

- The only exceptional case when cmderr does not reflect the outcome of the completed command is precisely described right in the very next paragraph. So as long as the sentence is not taken out of context, there should be no confusion on the reader's part.
- The proposed formulation also does not fully describe the situation: The command may still succeed and cmderr be non-zero (busy).
- I could not figure out a more precise phrasing that would not make the text harder to grasp at the same time. (But if you have another proposal in mind, feel free to post it.)